sdk/src/main/java/software/amazon/awssdk/iot/AwsIotMqttConnectionBuilder.java

```diff
@@ -644,6 +646,14 @@ public AwsIotMqttConnectionBuilder withCustomAuthorizer(String username, String
         }
         if (authorizerSignature != null)
         {
+            if (!authorizerSignature.contains("%")) {
+                try {
+                    authorizerSignature = URLEncoder.encode(authorizerSignature, StandardCharsets.UTF_8.toString());
+                } catch (UnsupportedEncodingException uee) {
+                    throw new CrtRuntimeException(uee.toString());
+                }
+            }
+            
```

Comment:
Similar to aws-crt-nodejs, we can check for a case when `tokenKeyName` and `tokenValue` are set while `authorizerSignature` is empty.

Reply:
Done